lib/read-index.js

```diff
@@ -6,7 +6,7 @@ module.exports = readIndex;
 function readIndex(url, options) {
   url = url.replace(/\/$/, '')
   return request('GET', url, options).getBody('utf8').then(function (body) {
-    var pattern = /\d\d\d\d\-[a-z]+\.txt(?:\.gz)?/gi
+    var pattern = options.archiveUrlRegex || /\d\d\d\d\-[a-z]+\.txt(?:\.gz)?/gi
```

Comment:
`options` may be `null` or `undefined`, please add a check that `options` is truthy before attempting to read the archiveUrlRegex.